explorer/explorertypes.go

```diff
@@ -574,17 +573,19 @@ func SyncStatusUpdate(from, to, timeToComplete int64, updateType, msg string) {
 
 	val := &SyncStatusInfo{
 		PercentComplete: percentage,
-		Msg:               msg,
-		Time:              timeToComplete,
-		UpdateType:        updateType,
+		Msg:             msg,
+		Time:            timeToComplete,
+		UpdateType:      updateType,
 	}
 
 	if len(blockchainSyncStatus.Update) == 0 {
-		blockchainSyncStatus.Update = map[string]*SyncStatusInfo{
-			updateType: val,
-		}
+		blockchainSyncStatus.Update = map[string]*SyncStatusInfo{updateType: val}
 	} else {
-		blockchainSyncStatus.Update[updateType] = val
+		if _, ok := blockchainSyncStatus.Update[updateType]; ok {
+			blockchainSyncStatus.Update[updateType] = val
+		} else {
+			blockchainSyncStatus.Update = map[string]*SyncStatusInfo{updateType: val}
```

Comment:
I added the `updateType` as a field in `SyncStatusInfo` since javascript does not support maps data structure. To send this data to the javascript side via the websocket, I have to convert the map into an array and the update type is one of the things that are needed to display data accurately and in the right position.

It so happens that looping through maps in golang doesn't maintain the items insertion order. Using the `updateType` as a value in the struct, we can ensure what progress data that should appear first will always appear as first on the status page.

Reply:
Let's chat in slack about this.